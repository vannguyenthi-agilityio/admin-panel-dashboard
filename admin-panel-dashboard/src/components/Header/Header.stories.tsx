import type { Meta, StoryObj } from '@storybook/react';

// Components
import Header from './';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta : Meta<typeof Header> = {
  title: 'Components/Header',
  component: Header,
  tags: ['autodocs'],
  argTypes: {
    user: { description: "Info of user" },
  },
};

export default meta;

type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Default: Story = {
  args: {},
};
