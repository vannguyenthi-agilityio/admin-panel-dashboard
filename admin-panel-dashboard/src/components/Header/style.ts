import { cva } from 'class-variance-authority';

export const headerStyles = cva(
  'w-full flex align-center gap-2 font-primary font-light text-sm bg-transparent placeholder:text-slate-400 text-slate-700 text-sm px-3 py-2 transition duration-300 ease focus:outline-none appearance-none cursor-pointer',
  {
    variants: {
      variant: {
        primary: ["bg-transparent", "text-dark"],
        secondary: ["bg-gray-100", "text-darker"],
      },
      size: {
        xs: ["text-xs", "px-2", "py-0"],
        sm: ["text-sm", "px-2", "py-2"],
        md: ["text-md", "px-2", "py-2"],
        lg: ["text-lg", "px-2", "py-2"],
      },
    },
    defaultVariants: {
      variant: "primary",
      size: "sm",
    },
  }
);
