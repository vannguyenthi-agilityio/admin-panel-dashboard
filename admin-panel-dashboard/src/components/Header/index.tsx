import { memo, forwardRef } from "react";

// Types
import { IHeader } from "@/types";

// Icons
import { MailIcon, BellIcon } from "@/components/Icons";

// Helpers
import { clsxMerge } from '@/helpers';

// Components
import { 
  Dropdown,
  Button
} from '@/components'

import { headerStyles } from './style';

const Header = forwardRef<HTMLDivElement, IHeader>(
  ({ 
    className,
    size,
    ...props }, ref) => (
    <div
      aria-label="Header"
      ref={ref}
      className={clsxMerge(
        headerStyles({ 
          size,
        }),
        className
      )}
      {...props}
    >
      <Button
        leftIcon={<BellIcon className="w-[16px] h-[16px]"/>}
        rightIcon={<div className="absolute block w-3 h-3 bg-red-500 border-2 border-white rounded-full top-1 right-2 dark:border-gray-900" />}
        className="px-2.5 h-[40px] relative"
      />
      <Button
        leftIcon={<MailIcon className="w-[16px] h-[16px]"/>}
        className="px-2.5 h-[40px]"
      />
      <Dropdown />
    </div>
  )
);

Header.displayName = "Header";
export default memo(Header);
