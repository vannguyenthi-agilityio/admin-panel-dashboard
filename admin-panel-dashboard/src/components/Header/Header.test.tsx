import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom';

// Components
import Header from '.'

describe('Header component', () => {
  it('Renders header default component', () => {
    const {container} = render(<Header />)
    expect(container).toMatchSnapshot();
  });
})
