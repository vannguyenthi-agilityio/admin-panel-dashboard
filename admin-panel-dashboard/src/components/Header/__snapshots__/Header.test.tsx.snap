// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Renders header default component 1`] = `
<div>
  <div
    aria-label="Header"
    class="flex align-center gap-2 font-primary font-light placeholder:text-slate-400 transition duration-300 ease focus:outline-none appearance-none cursor-pointer bg-transparent text-dark text-sm px-2 py-2"
  >
    <button
      class="flex flex-row gap-x-4 font-primary uppercase font-medium disabled:cursor-not-allowed items-center justify-center bg-white text-darker hover:bg-greyish border hover:border-greyish border-greyish hover:shadow-md text-sm py-2 rounded-lg px-2.5 h-[40px] relative"
      type="button"
    >
      <svg
        class="w-[16px] h-[16px]"
        fill="none"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 12.75C5 12.75 5 15.25 8 15.25C11 15.25 11 12.75 11 12.75M13.25 8V6C13.25 3.1005 10.8995 0.75 8 0.75C5.10051 0.75 2.75 3.10051 2.75 6V8L0.75 12.25H15.25L13.25 8Z"
          stroke="#212529"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
      <div
        class="absolute block w-3 h-3 bg-red-500 border-2 border-white rounded-full top-1 right-2 dark:border-gray-900"
      />
    </button>
    <button
      class="flex flex-row gap-x-4 font-primary uppercase font-medium disabled:cursor-not-allowed items-center justify-center bg-white text-darker hover:bg-greyish border hover:border-greyish border-greyish hover:shadow-md text-sm py-2 rounded-lg px-2.5 h-[40px]"
      type="button"
    >
      <svg
        class="w-[16px] h-[16px]"
        fill="none"
        height="14"
        viewBox="0 0 16 14"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.5 1.5L8 7.25L14.5 1.5M0.75 2.75C0.75 1.64543 1.64543 0.75 2.75 0.75H13.25C14.3546 0.75 15.25 1.64543 15.25 2.75V11.25C15.25 12.3546 14.3546 13.25 13.25 13.25H2.75C1.64543 13.25 0.75 12.3546 0.75 11.25V2.75Z"
          stroke="#212529"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </button>
    <div>
      <button
        class="font-primary border focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-center inline-flex items-center bg-white text-dark text-sm px-2 py-1.5"
        data-dropdown-toggle="dropdown"
        type="button"
      >
        <img
          alt="user photo"
          class="w-7 h-7 mr-2 rounded-lg"
          src="https://www.gravatar.com/avatar/2c7d99fe281ecd3bcd65ab915bac6dd5?s=250"
        />
        <span
          class="mr-2"
        >
          Derek Alvarado
        </span>
        <svg
          class="mx-2"
          fill="none"
          height="5"
          viewBox="0 0 8 5"
          width="8"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.25 0.75L4 4.25L0.75 0.75"
            stroke="#adb5bd"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="1.5"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
